x/auth/keeper/bech32_codec.go

```diff
@@ -0,0 +1,46 @@
+package keeper
+
+import (
+	"github.com/cosmos/cosmos-sdk/types/bech32"
+	"github.com/cosmos/cosmos-sdk/types/address"
+    sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+type bech32Codec struct {
```

Comment:
I have declared `bech32Codec` only to resolve the issue of equal names of address codec and rpc methods. Since @amaurym requested to change address codec names, I dont think we have to keep `bech32Codec`. I would remove `bech32Codec` and let  `authz AccountKeeper` implement `address.Codec` interface ( `StringToBytes` and `BytesToString` methods).

Reply:
I think having `bech32Codec` is actually clearer

Reply:
Sounds good